sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_asset_utils.py

```diff
@@ -146,25 +139,21 @@ def test_symlinks_included_in_hash(self, storage_test_directory: str) -> None:
         updated_hash = get_object_hash(path=link_file_path, ignore_file=no_ignore_file)
         assert original_hash != updated_hash
 
-    def test_symlink_upload_paths(self, storage_test_directory: str) -> None:
+    def test_symlink_upload_paths(self, storage_test_directory: str, fake_remote_prefix: str) -> None:
         """Confirm that symlink name is preserved for upload to storage, but that target file's path is uploaded
 
-        e.g given a file ./dir/foo/bar.txt with a symlink ./other_dir/bar_link.txt, we want to upload the contents of ./dir/food/bar.txt at path ./other_dir/bar_link.txt in the remote storage.
+        e.g given a symlink "./other_dir/bar_link.txt" with target file "./dir/foo/bar.txt", we want to upload the
```

Comment:
In this case, `./` is the upload directory?

Reply:
Also, how are you creating `other_dir`. `generate_link_file` seems to be creating both link and target file in the same folder?

Reply:
Yes, it's the upload directory.

yep - they're in the same folder; there is no "other_dir" in the actual test. That docstring was just an explanation of symlink/target and how the file name should be processed for remote upload with the "other_dir/bar_link.txt" -> "dir/foo/bar.txt" being a fake example. Would you want me to add another test case with the target and symlink in different directories?

Reply:
Yeah, that will be great. It can be in another PR though.

Reply:
note: work item has been created for this test case